assets/js/components/Dashboard/NewPaMessage/StationsAndZones/SelectZonesPage.tsx

```diff
@@ -140,35 +139,31 @@ const SelectZonesPage = ({
     );
   });
 
-  const screensByZone = useMemo(() => {
-    const screensWithAZone = allScreens.filter((s) => s.zone);
-    return fp.groupBy((s) => s.zone, screensWithAZone);
-  }, [allScreens]);
-
-  const areGroupsSelected = (groups: Array<ReadonlyArray<string>>) =>
-    groups.every((group) =>
-      group.every((zone) => {
-        return signIDs(screensByZone[zone] ?? []).every(isSelected);
-      }),
-    );
+  const isDirectionGroupSelected = (directionId: 0 | 1) =>
+    signIDs(
+      allScreens.filter((s) =>
+        s.routes?.map((r) => r.direction_id).includes(directionId),
+      ),
+    ).every(isSelected);
 
-  const selectGroups = (groups: Array<ReadonlyArray<string>>) => {
+  const selectDirectionGroup = (directionIds: (0 | 1)[]) => {
     const signsWithoutSelectedRoute = fp.without(signIDs(allScreens), value);
 
-    const newScreens = groups.flatMap((group) =>
-      group.flatMap((zone) => signIDs(screensByZone[zone] ?? [])),
+    const newScreens = allScreens.filter(
+      (s) =>
+        fp.intersection(
+          s.routes?.map((r) => r.direction_id),
+          directionIds,
+        ).length > 0,
```

Comment:
I think this might be easier to understand, and possibly more efficient, as:

```js
allScreens.filter((screen) =>
  screen.routes?.some((route) => directionIds.includes(route.direction_id))
)
```

Reply:
This was simplified as a result of the change [here](https://github.com/mbta/screenplay/pull/454#discussion_r1722245143).